### DISCUSSION TITLE
Tab stops, how are they set, can they be edited in any way?

### DISCUSSION THREAD

**Discussion by aleith** [glushchenko/fsnotes#1772] [General]
I've looked in the macOS FSNotes Settings for tab stops. I found a setting that substitutes spaces, like a code editor would.  I would like something a bit more than substitution of 2 or 4 spaces. Also searched these forums for Tab etc and not seen anything so far.

For my notes I'm using a tab between page references and content. It seems that the tab stop is set to some kind of measurement interval that just repeats across the page. As i'm using a non mono-spaced font, it's not possible to say what the tab stop is in chars or spaces (approx. 11 spaces). see screenshot below.
<img width="271" alt="Screenshot 2024-09-30 at 6 32 38 pm" src="https://github.com/user-attachments/assets/8bd365b9-1005-4131-a7d2-362e5ed04388">

I need it to be a little bit more than it is, is there a way to do that, even if it means hacking a .json or plist file? 

If I could have tabstops that indent all the text on the following lines until a \r r \n is encountered that would be ideal.  You can see the line beginning "S7" wraps to the start of the next line, indenting it consistently like some text editors do is what I'm trying to describe here.

___ 
Hello to the devs and user community.  I was thinking how I need an Apple Notes replacement for months now and then @fsnotes liked a completely unrelated post of mine on Twitter and so I became aware of this application. Funny how these things happen! I'm totally new to it but have read the intro texts a few times now. 

I'm very pleased to have discovered this app, and if I can use it on my iPhone and iPad  also, and sync across devices, I'll buy those versions to help support the development of FSNotes and adding new features.

**Comment by glushchenko** [comment]
It can't be changed, no changes are planned.

**Reply by aleith** [reply]
hi @glushchenko and thanks for improving on Apple's Notes app and making it OSS!

I'm wondering what workarounds I can consider using. For instance if I use manual space characters to approximate a Tab stop set at some distance from let margin, the 'indent' is visible work in "Draft?" mode but not in Preview mode as white space gets collapsed to a single space in Preview mode.

**Draft(?) mode &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; Preview mode**
<img width="201" alt="Screenshot 2024-10-25 at 6 10 39 pm" src="https://github.com/user-attachments/assets/8b765640-ec85-4348-be65-8d0a5a519a75">    <img width="203" alt="Screenshot 2024-10-25 at 6 10 55 pm" src="https://github.com/user-attachments/assets/0ecd0a79-8488-443b-a45a-87afa0264b83">

It seems like FSNotes will convert a tab keystroke into space chars on the fly if set to 2 or 4 Spaces in Settings, but all tab chars, and spaces get collapsed HTML style to a single space cha in Preview mode, regardless.
<img width="243" alt="FSNOTES settings tab stops " src="https://github.com/user-attachments/assets/4d8902a0-0e32-4add-a775-4a3207655ebb">

Far as I can tell no TextEdit style formatting controls that would allow for a negative "First Line" indent, or left and right indents for that matter, which would be the standard text editor and DTP type way of handling this formatting desire (shown below in TextEdit and InDesign).

**Comment by aleith** [comment]
I can think of a couple of ways of implementing this feature/facility. Is it worth my time making a Feature Request for it or is this of no interest at this time?

One would require simply having the option in FSNotes to prevent horizontal white space characters collapsing to a single space char in Preview mode. The other way would be the way GUI text editors and DTP software do it, much more work. I'm guessing the former would be easier to implement, more consistent with FSNotes "style" and for that reason possibly preferable to existing users.

In a code editor like Sublime.app I'd set the tap stop as equivalent to `4 spaces` and check "`View (menu) > Indentation > Indent using spaces`". That method allows quick input of space chars (in multiples of four if that's the tab stop)  and add/remove 1-3 spaces to always hit the desired alignment quickly (given a monospaced font of course 😊). 

White space get used **a lot** in Haskell language so I'm influenced by that, I suppose. 

White-space has syntactical meaning in some contexts for continuation of blocks/expressions/lines and to 'prettify' multiline expressions. 

Tab keystrokes are used to enter multiple spaces at the start of lines for syntax reasons. 

Long multi-space is used for indentation purposes at start of lines but are also _within_ lines frequently to 'prettify' expressions that appear in blocks. See the indenting to align the equals signs (`=`) and "is of type" (`::`) in code in screenshot. It's all done using spaces and they are not inconvenient to enter at all with automatic-indenting in software and tabs as _n_ spaces settings . 

<img width="1081" alt="Screenshot 2024-10-25 at 6 14 24 pm" src="https://github.com/user-attachments/assets/673edaf9-598b-4da6-a9f3-bff32235ed7b">

The  text is highlighted, we you can see the space chars have been used (as opposed to Tabs which show in Sublime as underscore lines Tabs chars). This is helpful. The faint lines show indicative `tab marks` (they aren't tab-stops) when text isn't highlighted.

Sublime supports this at the GUI level with menu items, but this isn't necessary, the existing `FSNotes > Settings…` would work with no changes required there for basic implementation of this formatting tweak. It's really just a code enhancement for the prevention of horizontal white space collapsing, HTML style, and possibly adding a Settings check mark to enable/disable it.

**Reply by glushchenko** [reply]
https://github.com/glushchenko/fsnotes/issues/1708
